Packages/blueberry-jam-core/Runtime/LevelLoader/SceneTransitioner.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace BJ
+{
+    public class SceneTransitioner
+    {
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSceneLoad)]
+        public static void Init()
+        {
+            // This config prefab is holds settings that allow th user to customize how the scene transitioner works for them.
+            // e.g. What sprites (that communicates to the player that a level is loading) are displayed during scene transitions? How long does fading to black and lifting blackout take? etc...
+            GameObject scene_transitioner_config = Resources.Load<GameObject>("SceneTransitionerConfig");
```

Comment:
This seems weird, maybe it should be a scriptable object here instead of a prefab as a file?

Reply:
As I understand it, the primary benefit of Scriptable Objects is that they reduce memory footprint as that are spawned once for all GameObjects that use them. Furthermore, they are more memory efficient, they persist across scenes meaning that they do not have to be consistently loaded. Is this correct?

Given that I am only loading one prefab does that not mean that the scriptable object would have a similar memory footprint to Scriptable Objects? Additionally, given that SceneTransitionerHelper is a singleton meaning that it is put in DONOTDESTROY, does that not mean that this prefab will be just as efficient as a scriptable object?

Reply:
When we import samples into projects, does the Resources folder get imported as well?

Reply:
Not right now, move the existing Resources folder into the BJSamples folder then it will work.
Everything in BJSamples is set up to be importable, the rest isn't.
This will not impact resource loading, LoadResource checks every folder called "Resources" including subfolders.